### PR TITLE
Bugfix Ticket 1175 - Added system tasks filter

### PR DESCRIPTION
**Pull request by eiresendez**
## Issue & Reproduction Steps
System tasks were being shown to users.

Reproduction Steps:
1. Install DocuSign package.
2. Create an application in DocuSign that has not been granted Access.
3. Go the next page.

image

4. Do not click "Authorize Access". Then go to `/tasks` and you should see the next records:

image

## Solution
- The `HideSystemResources` trait was used to exclude system-type records from the `process_request_tokens` table.

## How to Test
Please follow the "Reproduction Steps" of above. You should not see system tasks.

Run `vendor/bin/phpunit --filter testGetListNonSystemTasks tests/Feature/Api/TasksTest.php`

## Related Tickets & Packages
- [Ticket 1175](http://tickets.pm4overflow.com/tickets/1175)
- [FOUR-4808](https://processmaker.atlassian.net/browse/FOUR-4808)

## Code Review Checklist
- [x] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [x] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [x] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [x] This solution fixes the bug reported in the original ticket.
- [x] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [x] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [x] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [x] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [x] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-ticket-1175-4-2-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-ticket-1175-4-2-processmaker.ci.processmaker.net

File: resources/js/tasks/components/TasksList.vue [hunk 1]
```diff
@@ -376,7 +376,8 @@ export default {
                   "&per_page=" +
                   this.perPage +
                   filterParams +
-                  this.getSortParam(),
+                  this.getSortParam() +
+                  "&non_system=true",
```
Comment (nolanpro) [inline resources/js/tasks/components/TasksList.vue:380]: @eiresendez I think non-system should be the default. I think you can remove this and `->when(...` above and just use $query->nonSystem() by itself

**Review by nolanpro — CHANGES_REQUESTED** [review]

File: resources/js/tasks/components/TasksList.vue (hunk 1 above)
Comment (eiresendez) [reply resources/js/tasks/components/TasksList.vue:380]: @nolanpro I found that there were other places where that API endpoint was called and they needed to find the system tasks. For this reason, I set the scope only for /tasks (web)

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-ticket-1175-4-2-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-ticket-1175-4-2-processmaker.ci.processmaker.net

**Review by nolanpro — APPROVED** [review]

**Pull request merged**